src/components/AAPModal/AnsibleAutomationPlatformModal.tsx

```diff
@@ -0,0 +1,242 @@
+  const [status, setStatus] = React.useState<string>(
+    initialStatus ? initialStatus : 'provisioning',
+  );
+  const [ansibleUILink, setAnsibleUILink] = React.useState<string | undefined>();
+  const [ansibleUIUser, setAnsibleUIUser] = React.useState<string>();
+  const [ansibleUIPassword, setAnsibleUIPassword] = React.useState<string>('');
+  const [passwordHidden, setPasswordHidden] = React.useState(true);
+  const handleSetAAPCRError = (errorDetails: string) => {
+    setError(errorDetails);
+  };
+
+  const getAAPDataFn = React.useCallback(async () => {
+    try {
+      const signupData = await getSignupData();
+      if (signupData == undefined) {
+        setError('Unable to retrieve signup data.');
+        return;
+      }
+      const data = await getAAPData(signupData.defaultUserNamespace);
+      const status = getReadyCondition(data, handleSetAAPCRError);
+      setStatus(status);
+      if (data != undefined && data.items.length > 0 && data.items[0].status != undefined) {
+        // set UI link if available
+        if (data.items[0].status.URL) {
+          setAnsibleUILink(data.items[0].status.URL);
+        }
+        // set the user for accessing the UI
+        if (data.items[0].status.adminUser) {
+          setAnsibleUIUser(data.items[0].status.adminUser);
+        }
+        // set the password for the UI user
+        if (data.items[0].status.adminPasswordSecret) {
+          const adminSecret = await getSecret(
+            signupData.defaultUserNamespace,
+            data.items[0].status.adminPasswordSecret,
+          );
+          if (adminSecret != undefined && adminSecret.data != undefined) {
+            setAnsibleUIPassword(decode(adminSecret.data.password));
+          }
+        }
+      }
+    } catch (e) {
+      setError(errorMessage(e));
+    }
+  }, []);
+
+  React.useEffect(() => {
+    if (status === 'provisioning' || status === 'unknown' || status == '') {
+      const handle = setInterval(getAAPDataFn, SHORT_INTERVAL);
+      return () => {
+        clearInterval(handle);
+      };
+    }
+  }, [status]);
+
+  return (
+    <Modal
+      data-testid="aap-modal"
+      width={600}
+      aria-labelledby="aap-modal-title"
+      header={
+        // custom title to prevent ellipsis overflow on small screens
+        <Title id="aap-modal-title" headingLevel="h1">
+          {status === 'ready'
+            ? 'Ansible Automation Platform instance provisioned'
+            : 'Provisioning Ansible Automation Platform (AAP) instance'}
+        </Title>
+      }
+      isOpen
+      onClose={() => onClose()}
+    >
+      {error ? (
+        <Alert
+          title="An error occurred"
+          variant={AlertVariant.danger}
+          actionClose={<AlertActionCloseButton onClose={() => setError(undefined)} />}
+          isInline
+          className="pf-v5-u-mb-lg"
+        >
+          {error}
+        </Alert>
+      ) : null}
+      {(() => {
+        switch (status) {
```

Comment:
This one is quite long and not many state variables are used. I think the individual switch branches could be their own components. It will make it easier for the future maintenance.

Reply:
I agree, I've refactored things a bit and moved them in separate components, pls see https://github.com/RedHatInsights/developer-sandbox-ui/pull/61/commits/c721e10cc6e1fe7c9662372c0d4445b276a63d27